### PR TITLE
New Problem: Training Softmax Reg w/ Grad Descent

### PR DESCRIPTION
**Pull request by turkunov**
Softmax regression is an extension of logistic regression for 2 or more classes.

* Input:
  * X, y
  * learning_rate
  * iterations - n of epochs
* Output:
  * list of updated parameters
  * list of recorded losses (Cross Entropy) over iterations (starting from 0)

### CODE REVIEW

**Comment by turkunov** [comment]
Solves https://github.com/Open-Deep-ML/DML-OpenProblem/issues/284

**Comment by Open-Deep-ML** [comment]
make this question 105

**Comment by turkunov** [comment]
@Open-Deep-ML got it! Should be on par with other problems now

**Comment by turkunov** [comment]
just fixed a critical bug (https://discord.com/channels/1260668083031248967/1260668734746661017/1341844971359043635)

fyi @moe18

**Review by moe18 — APPROVED** [review]
Looks good
